components/brave_wallet/browser/blockchain_images_source_base.cc

```diff
@@ -0,0 +1,78 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/blockchain_images_source_base.h"
+
+#include <optional>
+#include <utility>
+
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/functional/bind.h"
+#include "base/memory/ref_counted_memory.h"
+#include "base/strings/string_util.h"
+#include "base/task/thread_pool.h"
+#include "brave/components/brave_wallet/browser/wallet_data_files_installer.h"
+
+namespace brave_wallet {
+
+BlockchainImagesSourceBase::BlockchainImagesSourceBase(
+    const base::FilePath& base_path)
+    : base_path_(base_path) {}
+
+BlockchainImagesSourceBase::~BlockchainImagesSourceBase() = default;
+
+void BlockchainImagesSourceBase::StartDataRequestForPath(
+    const std::string& path,
+    GotDataCallback callback) {
+  std::optional<base::Version> version =
+      brave_wallet::GetLastInstalledWalletVersion();
+  if (!version) {
+    std::move(callback).Run({});
+    return;
+  }
+
+  base::FilePath images_path(base_path_);
+  images_path = images_path.AppendASCII(version->GetString());
+  images_path = images_path.AppendASCII("images");
+  images_path = images_path.AppendASCII(path);
+
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE, {base::MayBlock()},
+      base::BindOnce(
+          [](const base::FilePath& path) {
+            return base::ReadFileToBytes(path);
+          },
+          images_path),
```

Comment:
It seems 
`base::BindOnce(&base::ReadFileToBytes, images_path),`
should work like here
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/webui/certificate_manager/user_cert_sources.cc;l=333